cloudnet-modules/cloudnet-cloudflare/src/main/java/de/dytanic/cloudnet/ext/cloudflare/CloudflareAPI.java

```diff
@@ -51,7 +51,7 @@ public static CloudflareAPI getInstance() {
         return CloudflareAPI.instance;
     }
 
-    public Pair<Integer, JsonDocument> createRecord(String serviceName, String email, String apiKey, String zoneId, DNSRecord dnsRecord) {
+    public Pair<Integer, JsonDocument> createRecord(String serviceName, String email, CloudflareConfigurationEntry.AuthenticationMethod authenticationMethod, String apiKey, String zoneId, DNSRecord dnsRecord) {
```

Comment:
The null check is missing here which will lead to problems if the authentication method is null

Reply:
Also you may mark them as @NonNull / @Nullable

Reply:
If I would do that here I would have to change it everywhere for consistency.

Reply:
I think it's better when it's your turn to do it straight anyway

Reply:
This PR just adds this feature, might clean up the Cloudflare module at a later point